core/src/test/java/org/opensearch/sql/expression/datetime/TimeStampDiffTest.java

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+package org.opensearch.sql.expression.datetime;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opensearch.sql.data.model.ExprDateValue;
+import org.opensearch.sql.data.model.ExprDatetimeValue;
+import org.opensearch.sql.data.model.ExprNullValue;
+import org.opensearch.sql.data.model.ExprStringValue;
+import org.opensearch.sql.data.model.ExprTimeValue;
+import org.opensearch.sql.data.model.ExprTimestampValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.DSL;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.ExpressionTestBase;
+import org.opensearch.sql.expression.FunctionExpression;
+import org.opensearch.sql.expression.function.FunctionProperties;
+
+class TimeStampDiffTest extends ExpressionTestBase {
+
+  private static Stream<Arguments> getTestDataForTimestampDiff() {
+    return Stream.of(
```

Comment:
You need to test extracting any type of diff from any types of values, for example, get year diff from two times (should be 0).

Reply:
I'm not sure if this is what you're referring to, but on line `124` of this file I have some tests below the comment saying `//Test with mismatched argument types`. Are these similar to the type of test you're asking me to make? I test a few mismatched types (example: first arg is `TIMESTAMP`, second arg is `STRING`), but I can add more tests to cover all cases.

Reply:
diff(day, time1, time2)
diff(day, time, date)
diff(hour, date1, date2)

Reply:
Reworked Test file in 1147b04b058f83b7c1f8040b4ad1795de81cce1b . The should help cover all ~250 permutations.